runtime/ts/storage/pouchdb/pouch-db-collection.ts

```diff
@@ -0,0 +1,364 @@
+import {CrdtCollectionModel, Model} from '../crdt-collection-model.js';
+import {Tracing} from '../../../../tracelib/trace.js';
+import {assert} from '../../../../platform/assert-web.js';
+import {PouchDbStorageProvider} from "./pouch-db-storage-provider.js";
+import {Type} from "../../type.js";
+import {PouchDbStorage} from "./pouch-db-storage";
+
+
+/**
+ * Defines a callback interface to allow for modifying a
+ * CrdtCollectionModel during a get/modify/set cycle.
+ */ 
+interface CrdtCollectionModelMutator {
+  (crdt: CrdtCollectionModel) : CrdtCollectionModel;
+}
+
+export class PouchDbCollection extends PouchDbStorageProvider {
+  /** The local synced model */
+  private _model: CrdtCollectionModel;  // NOTE: Private, but outside code accesses this :(
+
+  /**
+   * Create a new PouchDbCollection.
+   *
+   * @param type the underlying type for this collection.
+   * @param storageEngine a reference back to the PouchDbStorage, used for baseStorageKey calls.
+   * @param name appears unused.
+   * @param id see base class.
+   * @param key the storage key for this collection.
+   */
+  constructor(type: Type, storageEngine: PouchDbStorage, name: string, id: string, key: string) {
+    super(type, storageEngine, name, id, key);
+
+    this._model = new CrdtCollectionModel();
+    assert(this.version !== null);
+  }
+
+  /** @inheritDoc */
+  backingType() {
+    return this.type.primitiveType();
+  }
+
+  // TODO(lindner): appears unused
+  // clone() {
```

Comment:
It could just be that there's no test or recipe that exercises this currently.

Reply:
I think I found usage in shape.ts, uncommenting.  Added comments about adding tests.